src/DeviceState.hpp

```diff
@@ -0,0 +1,17 @@
+#ifndef POWER_WHISPERPOWER_DEVICESTATE_HPP
+#define POWER_WHISPERPOWER_DEVICESTATE_HPP
+
+namespace power_whisperpower {
+    /**
+     * @brief The state of the WhisperPower device
+     *
+     */
+    struct DeviceState {
+        DeviceState() = default;
+        DeviceState(bool running, bool stopped);
+        bool running = false;
+        bool stopped = false;
+    };
```

Comment:
Call it `GeneratorState`, and make the state an enum (ideally with the GENERATOR_STOPPED, GENERATOR_STARTING, GENERATOR_RUNNING and GENERATOR_STOPPING) states.

Reply:
- using an enum is less awkward since running and stopped cannot be true at the same time
- it opens the possibility of more information (like for instance the load % which I have the feeling we'll need)